R/arg.R

```diff
@@ -214,12 +210,79 @@ arg_text_ <- function(x, width = 60L, nlines = Inf) {
   paste0(str, collapse = "\n")
 }
 
-#' Generate a missing argument.
+#' Generate or handle a missing argument
 #'
+#' These functions help using the missing argument as a regular R
+#' object. It is valid to generate a missing argument and assign it in
+#' the current environment or in a list. However, once assigned in the
+#' environment, the missing argument normally cannot be
+#' touched. \code{maybe_missing()} checks whether the object is the
+#' missing argument, and regenerate it if needed to prevent R from
+#' throwing a missing error. In addition, \code{is_missing()} lets you
+#' check for a missing argument in a larger range of situations than
+#' \code{\link[base]{missing}()} (see examples).
+#' @param x An object that might be the missing argument.
 #' @export
 #' @examples
+#' # The missing argument can be useful to generate calls
 #' f_interp(~f(x = uq(arg_missing())))
 #' f_interp(~f(x = uq(NULL)))
+#'
+#'
+#' # It is perfectly valid to generate and assign the missing
+#' # argument.
+#' x <- arg_missing()
+#' l <- list(arg_missing())
+#'
+#' # Note that accessing a missing argument contained in a list does
+#' # not trigger an error:
+#' l[[1]]
+#' is.null(l[[1]])
+#'
+#' # But if the missing argument is assigned in the current
+#' # environment, it is no longer possible to touch it. The following
+#' # lines would all return errors:
+#' #> x
```

Comment:
I've had a CRAN maintainer run the dontrun examples in vdiffr and complain they were throwing errors. Maybe that was only because most of the examples were dontrun{}'d?